### PR TITLE
TINY-7908: SHY entities should not be treated as word-breaks

### PR DESCRIPTION
**Pull request by ashfordneil**
Related Ticket: TINY-7908 (take 2)

Description of Changes:
* Add soft-hyphens to the list of accepted word characters in Polaris
  * This affects Robin's zone-walker related APIs
  * This also affects the word count plugin

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/` for new features (if applicable)

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable): N/A

### CODE REVIEW

File: modules/polaris/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,9 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## Unreleased
 
+### Fixed
+- `Pattern.chars` and `Pattern.wordbreak` now accept soft hyphen (SHY) entities #TINY-7908
```
Comment (ashfordneil) [inline modules/polaris/CHANGELOG.md:10]:
```suggestion
- `Pattern.chars` and `Pattern.wordbreak` now treat soft hyphen characters (`&shy;` entities) as part of a word #TINY-7908
```

File: modules/robin/src/main/ts/ephox/robin/api/general/TextSearch.ts [hunk 2]
```diff
@@ -82,6 +82,7 @@ const expandRight = <E, D>(universe: Universe<E, D>, item: E, offset: number, ra
 // through until the offset left is 0. This is designed to find text node positions that
 // have been fragmented.
 const scanRight = <E, D>(universe: Universe<E, D>, item: E, originalOffset: number): Optional<SpotPoint<E>> => {
+  // MAKE THIS A LOOP
```
Comment (ashfordneil) [inline modules/robin/src/main/ts/ephox/robin/api/general/TextSearch.ts:85]:
```suggestion
```
This was a note I'd left for myself for a different task, please ignore

Comment (lnewson) [inline modules/robin/src/main/ts/ephox/robin/api/general/TextSearch.ts]: Why was this added as it doesn't appear to be related to this PR? If it should be added, then can we please make it a TODO (preferably with a Jira) instead?

File: modules/tinymce/CHANGELOG.md [hunk 3]
```diff
@@ -21,6 +21,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Fixed an exception getting thrown when the number of `col` elements didn't match the number of columns in a table #TINY-7041 #TINY-8011
 - As of Mozilla Firefox 91, toggling fullscreen mode with `toolbar_sticky` enabled would cause the toolbar to disappear #TINY-7873
 - Inserting content into a `contenteditable="true"` element that was contained within a `contenteditable="false"` element would move the selection to an incorrect location #TINY-7842
+- The `wordcount` plugin no longer treats soft hyphens (`&shy;` entities) as a word break #TINY-7908
```
Comment (lnewson) [inline modules/tinymce/CHANGELOG.md:24]: Should this be in the `Changed` section instead of the `Fixed` section?

**Review by lnewson — APPROVED** [review]

File: modules/robin/src/main/ts/ephox/robin/api/general/TextSearch.ts (hunk 2 above)
Comment (lnewson) [reply modules/robin/src/main/ts/ephox/robin/api/general/TextSearch.ts]: Ignore, you fixed this while I was reviewing :stuck_out_tongue:

**Review by ltrouton — APPROVED** [review]

File: modules/tinymce/CHANGELOG.md (hunk 3 above)
Comment (ashfordneil) [reply modules/tinymce/CHANGELOG.md:24]: I would say fixed because the old behaviour was a bug. Hyphens shouldn't have ever counted towards word breaks. Happy to change this if you feel strongly though

Comment (lnewson) [reply modules/tinymce/CHANGELOG.md:24]: I'm happy to leave it in fixed, but we'll need to reword this entry then, as this is very much so a `Changed` entry atm.

Comment (ashfordneil) [reply modules/tinymce/CHANGELOG.md:24]:
```suggestion
- The `wordcount` plugin was incorrectly treating soft hyphens (`&shy;` entities) as word breaks #TINY-7908
```

Comment (lnewson) [reply modules/tinymce/CHANGELOG.md:24]: LGTM :+1:

**Review by JamesToohey — APPROVED** [review]
